### DISCUSSION TITLE
problem with dirac-proxy-init

### DISCUSSION THREAD

**Discussion by zhangxiaomei** [DIRACGrid/DIRAC#7623] [Q&A] [answered]
A JUNO user successully registered her certificate in JUNO VOMS and DIRAC. From dirac server, the log of VOMS2CS has no errors with her registration. 
Her certificate works fine with voms-proxy-init --voms juno, but failed with "dirac-proxy-init -g juno_user"  

```
"2024-04-18 07:52:44 UTC dirac-proxy-init [140026756306752] WARN: Can't update from server Error while updating from dips://prod-dirac.ihep.ac.cn:9135/Configuration/Server: Peer closed connection
2024-04-18 07:52:45 UTC dirac-proxy-init [140026756306752] WARN: Reason(s):
    Peer closed connection
2024-04-18 07:52:45 UTC dirac-proxy-init [140026756306752] ERROR: Can't contact DIRAC CS: Reason(s):
    Peer closed connection"
```

More details in https://ggus.eu/index.php?mode=ticket_info&ticket_id=166266
What problem could it be?

**Comment by fstagni** [comment]
Can you post the CS section corresponding to this user?

**Reply by andresailer** [reply]
Also check the Configuration/Server log file if there is anything from the user trying to get a proxy.

**Comment by zhangxiaomei** [comment]
![图片](https://github.com/DIRACGrid/DIRAC/assets/1860588/eed6d837-5bd2-48de-8c46-71d49e0f316b)
![图片](https://github.com/DIRACGrid/DIRAC/assets/1860588/5d4a0875-052d-48ff-95e9-13c1a5d2565a)
She is correctly registered as a juno_user by the agent. I don't see any problems

**Comment by zhangxiaomei** [comment]
```
[dirac@prod-dirac startup]$ grep -ir "vzavadskyi" Configuration_Server/log/*
```
```
[dirac@prod-dirac startup]$ grep -ir "vzavadskyi" Configuration_VOMS2CSAgent/log/*
Configuration_VOMS2CSAgent/log/@40000000664bd1333b8b5ad4.s:2024-05-17 04:24:35 UTC Configuration/VOMS2CSAgent/VOMS2CSSynchronizer [140202118149952] INFO: Adding new user vzavadskyi: {'DN': '/C=CN/O=HEP/O=IHEP/OU=PHYS/CN=Vitalii Zavadskyi', 'CA': '/C=CN/O=HEP/CN=Institute of High Energy Physics Certification Authority', 'Email': 'zavadskyi@jinr.ru', 'Groups': ['juno_user']}
Configuration_VOMS2CSAgent/log/@40000000664bd1333b8b5ad4.s:2024-05-17 04:24:35 UTC Configuration/VOMS2CSAgent [140202118149952] INFO: Registering user vzavadskyi
Configuration_VOMS2CSAgent/log/@40000000664bd1333b8b5ad4.s:2024-05-17 04:24:35 UTC Configuration/VOMS2CSAgent [140202118149952] INFO: Added user vzavadskyi to group juno_user
Configuration_VOMS2CSAgent/log/@40000000664bd1333b8b5ad4.s:2024-05-17 04:24:35 UTC Configuration/VOMS2CSAgent [140202118149952] INFO: Registered user vzavadskyi
Configuration_VOMS2CSAgent/log/@40000000664bd1333b8b5ad4.s:  Added new user vzavadskyi:
```

**Reply by fstagni** [reply]
The user seems correctly registered, but why can't we see a log entry in the CS? It clearly says "connected". We need to see if this can be done again, "live".

One more question: is the certificate used a 2048 bit one? https://github.com/DIRACGrid/DIRAC/discussions/7310

**Reply by zhangxiaomei** [reply]
This reminds me that her cert could be the problem.  I only check her proxy, not her cert. I have asked her to check.
-bash-4.2$ voms-proxy-info
subject   : /C=CN/O=HEP/O=IHEP/OU=PHYS/CN=Vitalii Zavadskyi/CN=1747866808
issuer    : /C=CN/O=HEP/O=IHEP/OU=PHYS/CN=Vitalii Zavadskyi
identity  : /C=CN/O=HEP/O=IHEP/OU=PHYS/CN=Vitalii Zavadskyi
type      : RFC3820 compliant impersonation proxy
strength  : 2048
path      : /tmp/x509up_u22646
timeleft  : 11:59:50
key usage : Digital Signature, Key Encipherment

**Reply by zhangxiaomei** [reply, marked as answer]
Finally, found her certificate is 1024bit :(
It is the reason for the problem. Thank you for helping me debug out.
